crates/iceberg/src/spec/values.rs

```diff
@@ -996,10 +1039,552 @@ mod timestamptz {
+                                    if k == "key" {
+                                        key = Some(v);
+                                    } else if k == "value" {
+                                        value = Some(v);
+                                    }
+                                });
+                                match (key, value) {
+                                    (Some(k), Some(v)) => {
+                                        map.insert(
+                                            k.try_into(key_ty)?
+                                                .ok_or_else(|| invalid_err("list"))?,
+                                            v.try_into(value_ty)?,
+                                        );
+                                    }
+                                    _ => return Err(invalid_err("list")),
+                                }
+                            } else {
+                                return Err(invalid_err("list"));
+                            }
+                        }
+                        Ok(Some(Literal::Map(map)))
+                    }
+                    _ => Err(invalid_err("list")),
+                },
+                RawLiteralEnum::Record(Record {
+                    required,
+                    optional: _,
+                }) => match ty {
+                    Type::Struct(struct_ty) => {
+                        let iters: Vec<(i32, Option<Literal>, String)> = required
+                            .into_iter()
+                            .map(|(field_name, value)| {
+                                let field = struct_ty
+                                    .field_by_name(field_name.as_str())
+                                    .ok_or_else(|| invalid_err("record"))?;
+                                let value = value.try_into(&field.field_type)?;
+                                Ok((field.id, value, field.name.clone()))
+                            })
+                            .collect::<Result<_, Error>>()?;
+                        Ok(Some(Literal::Struct(super::Struct::from_iter(iters))))
+                    }
+                    Type::Map(map_ty) => {
+                        if *map_ty.key_field.field_type != Type::Primitive(PrimitiveType::String) {
+                            return Err(invalid_err("record"));
+                        }
+                        let mut map = BTreeMap::new();
+                        for (k, v) in required {
+                            map.insert(
+                                Literal::string(k),
+                                v.try_into(&map_ty.value_field.field_type)?,
+                            );
+                        }
+                        Ok(Some(Literal::Map(map)))
+                    }
+                    _ => Err(invalid_err("record")),
+                },
+                RawLiteralEnum::StringMap(_) => Err(invalid_err("string map")),
```

Comment:
Why this is invalid?

Reply:
>In avro, Value::Record and Value::Map both will call visitor::map
. So we convert them to RawLiteral::Record first and then according to the type info, we know convert it to Literal::Record or Literal::Map.

That's why it's invalid. Because in deserialize, the map will be converted to `RawLiteral::Record`. We can see in `RawLiteralVisitor` there is nothing will be convert to StringMap. Actually, the `StringMap` is only used in serialize.